factory/src/main/java/com/google/auto/factory/processor/AutoFactoryDeclaration.java

```diff
@@ -166,7 +175,54 @@ PackageAndClass getFactoryName() {
               implementingTypes,
               allowSubclasses,
               mirror,
-              ImmutableMap.copyOf(values)));
+              ImmutableMap.copyOf(values),
+              factoryAnnotations));
+    }
+
+    private List<CopyAnnotationsDescriptor> getFactoryAnnotations(Element element) {
+      
+      if( element instanceof ExecutableElement ) {
+        return getFactoryAnnotations(element.getEnclosingElement());
+      }
+      ImmutableList.Builder<CopyAnnotationsDescriptor> resultBuilder = ImmutableList.builder();
+      
+      CopyAnnotations fa = element.getAnnotation(CopyAnnotations.class);
```

Comment:
The auto-factory does have this direct reference to the annotation in 5 locations.
Perhaps this can be done in a different change in future, and then do it for all of those references?

Reply:
OK